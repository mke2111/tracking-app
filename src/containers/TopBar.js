import styled from 'styled-components';
import { useSelector } from 'react-redux';

const Header = styled.header`
background-color: #62b5e5;
padding: 1rem 2rem;
color: white;
display: flex;
justify-content: center;
align-items: center;`;

const TopBar = () => {
  const { tab } = useSelector(state => state);
  const headerText = tab.tabName;
  return (
    <Header>
      <h3>{headerText}</h3>
    </Header>
  );
};

export default TopBar;
