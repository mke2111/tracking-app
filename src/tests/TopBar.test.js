import {
  render, screen, act, waitFor,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
// import store from '../reducers/store';
import { createStore } from 'redux';
import TopBar from '../containers/TopBar';

const store = createStore(() => ({ tab: { tabName: 'Test' } }));

describe('Rendering component', () => {
  it('creates a TopBar component', async () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <TopBar />
          </Provider>
        </BrowserRouter>,
      );
    });
    await waitFor(() => {
      expect(screen.getByText('Test'));
    });
  });
});

describe('TopBar', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <TopBar />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
