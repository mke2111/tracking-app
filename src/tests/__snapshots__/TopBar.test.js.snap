// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders correctly 1`] = `
<header
  className="bg-blue-500 text-white text-center h-12 py-1 flex justify-between"
>
  <a
    href="https://github.com/mke2111/tracking-app"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="Github"
      className="w-8 h-8 ml-3 cursor-pointer"
      src="github.png"
    />
  </a>
  <h3
    className="pt-2"
  >
    Test
  </h3>
  <h3
    className="pt-2 invisible"
  >
    Nothing
  </h3>
</header>
`;
